File: src/WebKontrol.py
from puppet import webPuppet
from admin import webAdmin
from config import webConfig
import time

config = webConfig()

admin = webAdmin()
admin.start()

puppet = webPuppet()
puppet.update()




def tryUpdate(postvalue = None):
    
    postvalueNew = admin.getPostValue()

    if(postvalueNew == "" or postvalueNew != None):
        postvalue = postvalueNew

    elif(postvalue == "" or postvalue == None):
        return False

    puppet.set_url(postvalue)
    admin.set_url(puppet.get_url())
    config.set_url(puppet.get_url())
    config.save()


    if(puppet.update()):
        print("New URL: " + postvalue)

        return True

    else:
        postvalueNew = admin.getPostValue()            
        sleeptime = 30 + time.time()
        while (sleeptime > time.time()):
            time.sleep(0.1)
            postvalueNew = admin.getPostValue()
            if(postvalueNew != None):
                postvalue = postvalueNew
                break
        return tryUpdate(postvalue)



if config.check_config():
    config.load()
    print("Config found, URL: " + puppet.get_url())
    tryUpdate(config.get_url())
else:
    print("No config found.")
    tryUpdate(puppet.get_url())


while(1):
    try:
        time.sleep(0.1)

        tryUpdate()
    except KeyboardInterrupt:
        puppet.stop()
        admin.stop()
        break

print("Shutting down...")

File: src/src/WebKontrol.ts

import { SettingLoader } from "./SettingLoader";
import { WebServer } from "./WebServer";
import { Puppet } from "./Puppet";


export class WebKontrol {

    webServer: WebServer;
    puppet: Puppet;
    settingLoader: SettingLoader;

    constructor(chromiumLocation: string|undefined = undefined) {
        // super();
        this.settingLoader = new SettingLoader();
        this.webServer = new WebServer(this.settingLoader);
        this.puppet = new Puppet(chromiumLocation);


        this.settingLoader.on('loaded', (config) => 
        {
            this.webServer.start();
            this.puppet.init().then(() =>
            {
                this.puppet.openPage(this.settingLoader.getSettings().url);
            });           
        });
      
        this.webServer.on('reload', () => {
            this.puppet.openPage(this.settingLoader.getSettings().url);
            console.log("Puppet reloaded!");
        });
      
        this.settingLoader.load();

        this.settingLoader.on('set_url', (value: string) => {
            console.log("Url Set!", value);
            this.puppet.openPage(value);
        });
    }

}








